src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -554,41 +561,35 @@ void recordWriteAfterQueryExecution() throws SQLException {
 
     public Statement getReadStatement(RouteDecisionBuilder decisionBuilder) {
         if (connectionProvider.getState().equals(MAIN)) {
-            decisionBuilder.reason(MAIN_CONNECTION_REUSE);
+            decisionBuilder.reason(MAIN_CONNECTION_REUSE);//.isWrite(true) ? ;
+            // sth was readonly, but need to happen on Main because of inconsistency
```

Comment:
The `isWrite` check needs to happen before the `isConsistent` check

Reply:
that depends,
I guess i should narrowed it down

- [ ]  `isWrite` -> `isSQLWrite`
- [ ]  `isWrite` -> `wasItTreatedLikeWrite` 

Then it answers whether consistency should be 1 of the factors to build it.

Reply:
@dagguh 
I've checked and `isConsistent` happens at the end of this method in `setCurrentStatement` when all reasons to prepareWriteStatement have been filter out, and still Replica is in the game for given query or rather statement.

Reply:
Nice, so maybe `Reason.REPLICA_INCONSISTENT` is not hiding anything and maybe #50 can be solved with the current API?

Reply:
I'm lost. Can you explain?

Reply:
It seems there is no actual case of `Reason.REPLICA_INCONSISTENT` hiding a write.

Reply:
We should ensure the contract is covered by a test.

Reply:
If `Reason`s already have all the information, then we don't need to rewire the internals. But we should still provide new API from #48 to keep the semantics local. E.g. in case we add a new reason or we decide to change the internal contract.

Reply:
I've pushed new changes, it looks like REPLICA_INCONSISTENT and MAIN_CONNECTION_REUSE contain information that isSQLWrite= false ( I've introduced isSQLWrite to enforce split between consistency and isWrite) for this Reasons are enough, though it wasn't obvious at the first glance)